makefile

```diff
@@ -314,7 +314,8 @@ endif
 
 # Source dirs in logical order
 DIRS = general linalg mesh fem
-SOURCE_FILES  = $(foreach dir,$(DIRS),$(wildcard $(SRC)$(dir)/*.cpp))
+KERNEL_DIRS  = general/kernels linalg/kernels mesh/kernels fem/kernels
```

Comment:
Could DIRS be populated with $(dir)/kernels like the `SOURCE_FILES`? That would allow a one-line modification.

Reply:
I don't think I quite follow. Doing something like 
TOP_DIRS = general linalg mesh fem
DIRS = $(TOP_DIRS) $(TOP_DIRS)/kernels
Doesn't quite work.

Reply:
I think he means
```
KERNEL_DIRS = $(DIRS:=/kernels)
```
... or `patsubst(%,%/kernels,$(DIRS))` works too.

Reply:
I see, yup. Using ```KERNEL_DIRS = $(DIRS:=/kernels)`` does the job too 👍